packages/demo-cards/claim.gts

```diff
@@ -55,90 +52,24 @@ class Isolated extends Component<typeof Claim> {
 
   constructor(owner: unknown, args: any) {
     super(owner, args);
-    this.initialize.perform();
-    if (window.ethereum) {
-      window.ethereum.on('chainChanged', (chainId: string) => {
-        this.connected =
-          parseInt(chainId, 16) == this.args.model.chain?.chainId;
-      });
-    }
-  }
-  private initialize = enqueueTask(async () => {
-    let isSameNetwork = this.isSameNetwork();
-    let isConnected = await this.isMetamaskConnected();
-    this.connected = isConnected && isSameNetwork;
-  });
-
-  isSameNetwork() {
-    let metamaskChainId = this.getChainId();
-    return this.args.model.chain?.chainId == metamaskChainId;
-  }
-
-  isMetamaskInstalled() {
-    return window.ethereum !== 'undefined';
+    this.chainId = this.args.model.chain?.chainId ?? -1;
+    this.metamask = new MetamaskResource({
+      named: { chainId: this.chainId },
+    }).setup();
```

Comment:
This resource looks probably unable to react to changes in its inputs. Usually resources are constructed with a closure argument so that they can consume tracked properties.